internal/argocd/health.go

```diff
@@ -136,6 +148,9 @@ func (h *applicationHealth) GetApplicationHealth(
 		}
```

Comment:
In my case, the information these fine-grained details added was quite necessary when diagnosing what is broken and why. I am convinced that debugging is the right level of setting for such, but I'm not against setting some to trace level. 

That however would go against the expectation people usually have when wanting to see "all" logs. Debug is the most verbose in most popular projects, few have a distinct "trace" setting, so I wouldn't think to look for that.

I think that having detailed logging for such a complex machinery is necessary, and as time goes by and the app matures and evolves it will only become more important.

Please advise.